src/modules/Lan.rb

```diff
@@ -1088,6 +997,68 @@ def activate_network_service
         LanItems.reload_config(ifaces)
       end
     end
+
+    def configure_as_bridge!(ifcfg, bridge_name)
+      return false if !NetworkInterfaces.Edit(ifcfg)
+
+      old_config = deep_copy(NetworkInterfaces.Current)
+      log.debug("Old Config #{ifcfg}\n#{old_config}")
+
+      log.info("old configuration #{ifcfg}, bridge #{bridge_name}")
+
+      NetworkInterfaces.Name = bridge_name
+
+      # from bridge interface remove all bonding-related stuff
+      NetworkInterfaces.Current.each do |key, _value|
+        NetworkInterfaces.Current[key] = nil if key.include? "BONDING"
+      end
+
+      NetworkInterfaces.Current["BRIDGE"] = "yes"
+      NetworkInterfaces.Current["BRIDGE_PORTS"] = ifcfg
+      NetworkInterfaces.Current["BRIDGE_STP"] = "off"
+      NetworkInterfaces.Current["BRIDGE_FORWARDDELAY"] = "0"
+
+      # hardcode startmode (bnc#450670), it can't be ifplugd!
+      NetworkInterfaces.Current["STARTMODE"] = "auto"
+      # remove description - will be replaced by new (real) one
+      NetworkInterfaces.Current.delete("NAME")
+      # remove ETHTOOLS_OPTIONS as it is useful only for real hardware
+      NetworkInterfaces.Current.delete("ETHTOOLS_OPTIONS")
+
+      NetworkInterfaces.Commit
+    end
+
+    def propose_current_item!(config)
+      # first configure all connected unconfigured devices with dhcp (with default parameters)
+      hwinfo = config.fetch("hwinfo", {})
+
+      if hwinfo.fetch("link", false) == true
+        log.warn("item number #{LanItems.current} has link:false detected")
+
+        return false
+      end
+
+      if hwinfo.fetch("type", "") == "wlan"
+        log.warn("not proposing WLAN interface")
+
+        return false
+      end
+
+      LanItems.ProposeItem
+    end
+
+    def refresh_lan_items
```

Comment:
I haven't changed the current behavior just make it more readable, with a little cleanup. As commented, this is a first step just for what are you suggesting, which things are still valid, which could be removed/reimplemented and which needs a better implementation. 

I just wanted to fix the original issue with a small cleanup, but opening the discussion for improvements.